cowprotocol/accounting/rewards/mainnet/processed_token_imbalances_4057345.sql

```diff
@@ -0,0 +1,352 @@
+        contract_address as token,
+        cast(shares as int256) as amount_wei,
+        'AMM_IN' as transfer_type
+    from batch_meta as bm
+    inner join maker_ethereum.SavingsDai_evt_Withdraw as w
+        on bm.tx_hash = w.evt_tx_hash
+    where owner = 0x9008d19f58aabd9ed0d60971565aa8510560ab41
+    union all
+    -- deposit events result in additional AMM_OUT transfer
+    select
+        tx_hash,
+        0x0000000000000000000000000000000000000000 as sender,
+        0x9008d19f58aabd9ed0d60971565aa8510560ab41 as receiver,
+        contract_address as token,
+        cast(shares as int256) as amount_wei,
+        'AMM_OUT' as transfer_type
+    from batch_meta as bm
+    inner join maker_ethereum.SavingsDai_evt_Deposit as w
+        on bm.tx_hash = w.evt_tx_hash
+    where owner = 0x9008d19f58aabd9ed0d60971565aa8510560ab41
+),
+
+all_transfers_temp as (
+    select * from user_in
+    union all
+    select * from user_out
+    union all
+    select * from other_transfers
+    union all
+    select * from eth_transfers
+    union all
+    select * from sdai_deposit_withdrawal_transfers
+),
+
+pre_batch_transfers as (
+    select * from all_transfers_temp
+    order by tx_hash
+),
+
+batch_transfers as (
+    select
+        block_time,
+        block_number,
+        pbt.tx_hash,
+        dex_swaps,
+        num_trades,
+        solver_address,
+        sender,
+        receiver,
+        token,
+        amount_wei,
+        transfer_type
+    from batch_meta as bm
+    inner join pre_batch_transfers as pbt
+        on bm.tx_hash = pbt.tx_hash
+),
+
+incoming_and_outgoing_temp as (
+    select
+        block_time,
+        tx_hash,
+        dex_swaps,
+        solver_address,
+        transfer_type,
+        case
+            when t.symbol = 'ETH' then 'WETH'
+            when t.symbol is not null then t.symbol
+            else cast(i.token as varchar)
+        end as symbol,
+        case
+            when token = 0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
+                then 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2
+            else token
+        end as token,
+        case
+            when receiver = 0x9008d19f58aabd9ed0d60971565aa8510560ab41
+                then amount_wei
+            when sender = 0x9008d19f58aabd9ed0d60971565aa8510560ab41
+                then cast(-1 as int256) * amount_wei
+        end as amount
+    from batch_transfers as i
+    left outer join tokens.erc20 as t
+        on
+            i.token = t.contract_address
+            and blockchain = 'ethereum'
+),
+
+incoming_and_outgoing as (
+    select
+        block_time,
+        tx_hash,
+        solver_address,
+        symbol,
+        token,
+        amount,
+        transfer_type
+    from incoming_and_outgoing_temp
+    order by block_time
+),
+
+-- add correction for protocol fees
+raw_protocol_fee_data as (
+    select
+        order_uid,
+        tx_hash,
+        cast(cast(data.protocol_fee as varchar) as int256) as protocol_fee,
```

Comment:
If there is no other way, you can use `-- noqa: RF01` to deactivate a specific rule on a line, or `--noqa` to deactivate all rules